.github/workflows/valgrind.yml

```diff
@@ -0,0 +1,81 @@
+name: Valgrind
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        description: "OS"
+        type: string
+        required: true
+      architecture:
+        description: "OS Architecture"
+        type: string
+        required: true
+      environment:
+        description: "Valgrind Environment"
+        type: string
+        required: true
+
+  workflow_dispatch:
+    inputs:
+      os:
+        description: "OS"
+        type: choice
+        options:
+          - alt:sisyphus
+          - alt:p10
+          - alt:p9
+        required: false
+        default: "alt:sisyphus"
+      architecture:
+        description: "OS Architecture"
+        type: choice
+        options:
+          - i386
+          - amd64
+        required: false
+        default: "amd64"
+      environment:
+        description: "Valgrind Environment"
+        type: choice
+        options:
+          - openldap
+          - samba
+        required: false
+        default: "openldap"
+
+jobs:
+  valgrind:
+    runs-on: ubuntu-latest
+    name: build
+
+    steps:
+    - name: Checkout
+    - uses: actions/checkout@v3
+      
+    - run: mkdir -p /tmp/certs
+
+    - run: export SCRIPT_PATH=`pwd` && cd /tmp/certs && bash "${SCRIPT_PATH}/script/generate_cert.sh" `pwd` && docker volume create --driver local --opt type=none --opt device=`pwd` --opt o=bind cert-volume
+
+    - run: docker volume create --driver local --opt type=none --opt device=`pwd` --opt o=bind data-volume
+
+    - run: cd tests && docker compose -f docker-compose.valgrind.${{ inputs.environment }}.yml -f docker-compose.yml build --build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g) --build-arg DISTR=${{ inputs.os }} --build-arg ARCH=${{ inputs.architecture }}
+
+    - run: cd tests && docker compose -f docker-compose.valgrind.${{ inputs.environment }}.yml -f docker-compose.yml up --abort-on-container-exit --attach-dependencies
+
+
+    # - name: Install valgrind and development dependencies
+    #   run: |
+    #     sudo apt-get update -y
+    #     sudo apt-get install -y cmake libc6-dev valgrind
+    #   run: |
+    #     gcc --version
+    #     mkdir build
+    #     cd build
+    #     cmake -DLIBDOMAIN_BUILD_TESTS:BOOL=ON -DCMAKE_BUILD_TYPE=Debug ..
+    #     make -j `nproc`
+    # - name: Test under valgrind
+    #   run: |
+    #     cd build/tests/auto
+    #     # TODO: valgrind -q --error-exitcode=99 --leak-check=full ./something
+    #     # TODO: docker-compose !!!
```

Comment:
While locally you can have such a comments. When doing commit or PR they should be removed.